### PR TITLE
Split abstract class to remove add link from child resources

### PR DESCRIPTION
**Pull request by sroberts2**
The child resources do not need the add link exposed or implemented, created an interface to model this.

Resolves: SFA-703

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/service/AbstractParentService.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.companieshouse.api.accounts.service;
+
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.model.rest.RestObject;
+
+public interface AbstractParentService<T extends RestObject> extends AbstractService<T> {
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/AbstractParentService.java]: The class name "AbstractParentService" suggests that this is an abstract class. I think the interface should be named what it is, a ParentService with a JavaDoc explaining what the interface is a contract for (it should be clear what the difference is between a service and a parent service without having to go looking at the code).

File: src/main/java/uk/gov/companieshouse/api/accounts/service/SmallFullService.java [hunk 2]
```diff
@@ -4,6 +4,6 @@
 import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
 
 @Service
-public interface SmallFullService extends AbstractService<SmallFull> {
+public interface SmallFullService extends AbstractParentService<SmallFull> {
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/SmallFullService.java]: Why is the CompanyAccountServiceImpl not also implementing the AbstractParentService? It also has an addLinks() method.

Comment (sroberts2) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/SmallFullService.java]: The CompanyAccount is a special case which could be made to fit the interface but doesn't fit the mould at the moment.

File: src/main/java/uk/gov/companieshouse/api/accounts/service/AbstractParentService.java (hunk 1 above)
Comment (sroberts2) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/AbstractParentService.java]: Agreed I will rename them.

**Pull request merged**